crates/sui-framework/packages/sui-framework/sources/event.move

```diff
@@ -37,6 +40,70 @@ module sui::event;
 /// phantom parameters, eg `emit(MyEvent<phantom T>)`.
 public native fun emit<T: copy + drop>(event: T);
 
+public use fun destroy_stream as EventStream.destroy;
+public use fun destroy_cap as EventStreamCap.destroy;
+public use fun emit_authenticated as EventStreamCap.emit;
+
+#[allow(unused_field)]
+public struct EventStreamHead has store {
+    /// Merkle root for all events in the current checkpoint.
+    root: vector<u8>,
+    /// Hash of the previous version of the head object.
+    prev: vector<u8>,
+}
+
+// A unique identifier for an event stream.
+public struct EventStream has key, store {
+    id: UID,
+}
+
+public fun new_event_stream(ctx: &mut TxContext): EventStream {
+    EventStream {
+        id: object::new(ctx),
+    }
+}
+
+public fun destroy_stream(stream: EventStream) {
```

Comment:
- What happens if you destroy a stream, but not its capability, and then emit to that capability?
- Should the stream track the number of capabilities that can write to it, and prevent itself from being destroyed if it can still be written to?
- Should destroying the stream be considered a privileged operation (protected by the capability?)

Reply:
All good questions. Nothing seems to be immediately harmed if there are orphaned capabilities. In fact the current API would allow you to ensure that a cap is unique forever by making one cap and destroying the EventStream (not sure if that would be useful).

I don't see any reason to protect destruction of the stream, it seems that taking the stream by value is enough of a restriction.